### ISSUE TITLE
Servers allow to set Limit concurrent connections

### ISSUE THREAD

**Issue by babobski** [Komodo/KomodoEdit#3313]
### Short Summary

In FileZilla and PHP storm you have the ability to set a connection limit in you're server settings.

![afbeelding](https://user-images.githubusercontent.com/3530262/31854453-ebfb14f2-b699-11e7-83de-ca3931635dfd.png)

Would be nice if this also comes to Komodo.

### Platform Information

Komodo IDE 11.0.1 Windows 10

**Comment by Naatan** [comment]
While I agree this would be nice this simply isn't going to be feasible for us as it would require a significant rewrite in logic.

**Comment by babobski** [comment]
Oke, read a post on the forum of someone hitting a max connection error with a third party program,
It just reminded me hitting the same issue a few times over the last couple years.

I think it would help out in those situations, if you could configure a max connection for you're server.
To give a example, last year we moved to a new hosting server, we have one server for development.
In our old environment we had set the max processes of the server to a higher value.

On the new server we where hitting the problem that this was not yet configured.
Causing the problem that some of our developers couldn't connect to the server any more.
The most of our developers are using PHP Storm and a solution that came forward, was to set the max connection to 2 for the development server.

A other solution was to switch to SFTP, this is the road I went.
Only then I was hitting problems like #1758 ( and the possible related #1782  ).
I know those issues will be fixed in the end, but when you encounter server problems, a max connection configuration can save you're day.

Now might not be the right moment, but I think it would be a good feature to have on you're road map.

**Comment by Naatan** [comment]
The problem with that ticket appears to be that either Komodo or the server (TBD) is not closing the connections properly once they have concluded their business. 

We'll definitely keep this in mind, but this would not be fixed as part of a bug. It would take a rewrite of some sort.

**Comment by babobski** [comment]
When I had those issues, I have been digging in the source code to find a solution.
Tried to close the existing connection on project close, but when I created the script's it only back fired for me (when i closed a connection a new one was made).

I would agree that implementing such function you have the rewrite a lot of logic's (as current it does not manege the connections)But if you keep it in mind, this was the only purpose of this ticket.
